src/mem/ruby/system/GPUCoalescer.cc

```diff
@@ -599,6 +599,10 @@ GPUCoalescer::hitCallback(CoalescedRequest* crequest,
                 // data response is not needed.
                 case RubyRequestType_ATOMIC_NO_RETURN:
                     assert(pkt->isAtomicOp());
+                    log = data.popAtomicLogEntryFront();
+                    delete [] log;
+                    data.setData(pkt->getPtr<uint8_t>(), offset, pkt_size);
```

Comment:
@BujSet is this necessary for ATOMIC_NO_RETURN?

Reply:
I think this would solve the assertion error if you need results quickly, however, this still only addresses the problem after the fact. I started a patch that will fix this issue by not creating the log when it's not needed (i.e. when the request is an AtomicNoReturn). I'm still working on this though since it touches quite a few different files (key issue is that the VIPER messages reuse the CPU message structure which ignores the type of atomic in the incoming RubyRequest, and in the event that the request is a miss the atomic action happens on the response path, so the delineation of what type of atomic request is issued must be preserved throughout the entire trip of the request). Will provide an update when I have a working solution.

Reply:
Got it, I'll wait for your update and take down this PR then. 
I can also help make those changes if you need extra hands; let me know

Reply:
See my PR here #546. @abmerop, When I was looking through the protocol code for the directories, I noticed that only the SLC bit is set and read. Rather than implementing checks for the GLC bit, I just added some extra types to distinguish the atomic request type. Do you think this would be fine, or would it be preferable to add in the support for the GLC bit set and checks?

Reply:
Since you've already done the work, I think it is cleaner not to pass the GLC bit beyond TCC as it isn't needed otherwise.  I thought it was already passed up hence my earlier comment

Reply:
Cool, I will close this PR for now. @BujSet I meant to ask if line 604 above is necessary for the NO_RETURN case; if not, you could add a break statement.

Reply:
Oh. I'm not sure. In line with the comment, I just followed the same format as what happens for a store. But thinking about more, maybe setData doesn't need to be called for either store or atomic no return requests. Will look into this.